src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -22,56 +21,83 @@
-    // in the address book
-    private static final Phone dummyPhone = new Phone("12345678");
-    private static final Email dummyEmail = new Email("dummyemail@gmail.com");
-    private static final Address dummyAddress = new Address("Dummy Address");
-
     private final Name nameOfPersonToDelete;
+    private final Index targetIndex;
+    private final boolean isDeletionByIndex;
 
-
+    /**
+     * Constructs a DeleteCommand for deletion by name
+     * @param nameOfPersonToDelete The name of the person to be deleted
+     */
     public DeleteCommand(Name nameOfPersonToDelete) {
         this.nameOfPersonToDelete = nameOfPersonToDelete;
+        this.targetIndex = Index.fromOneBased(1); //a dummy value
+        this.isDeletionByIndex = false;
     }
 
+    /**
+     * Constructs a DeleteCommand for deletion by index
+     * @param targetIndex The index of the person to be deleted on the filtered list on GUI
+     */
+    public DeleteCommand(Index targetIndex) {
+        this.nameOfPersonToDelete = new Name("name"); //a dummy value
+        this.targetIndex = targetIndex;
+        this.isDeletionByIndex = true;
+    }
+
+
     @Override
     public CommandResult execute(Model model) throws CommandException {
 
         requireNonNull(model);
 
-        Person personWithNameToDelete = new Person(nameOfPersonToDelete);
-
-        if (!model.hasPerson(personWithNameToDelete)) { // model.hasPerson considers 2 Persons with same name
-            // to be the same Person
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_NAME);
-        }
-
-        ObservableList<Person> personList = model.getAddressBook().getPersonList();
+        if (isDeletionByIndex) { //deletion by index
+            List<Person> lastShownList = model.getFilteredPersonList();
+            if (targetIndex.getZeroBased() >= lastShownList.size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+            }
+            Person personToDelete = lastShownList.get(targetIndex.getZeroBased());
+            model.deletePerson(personToDelete);
+            return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, personToDelete));
 
-        final Person[] personToDelete = new Person[1];
+        } else { //deletion by name
+            Person personWithNameToDelete = new Person(nameOfPersonToDelete);
 
-        personList.forEach((person) -> {
-            if (person.isSamePerson(personWithNameToDelete)) {
-                personToDelete[0] = person;
-                model.deletePerson(person);
+            if (!model.hasPerson(personWithNameToDelete)) { //model.hasPerson considers 2 Persons with same name
+                //to be the same Person
+                throw new CommandException(Messages.MESSAGE_INVALID_PERSON_NAME);
             }
-        });
-
-        return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, personToDelete[0]));
+            //todo : make this more oop
```

Comment:
I agree, down the line we should probably have a separate model method that can return a `Person` object or list of them, given a name or index.